MemoryMeter.c

```diff
@@ -33,30 +33,17 @@ static void MemoryMeter_updateValues(Meter* this) {
    size_t size = sizeof(this->txtBuffer);
    int written;
 
-   Settings *settings = this->host->settings;
+   this->curItems = ARRAYSIZE(MemoryMeter_attributes);
 
    /* shared, compressed and available memory are not supported on all platforms */
    this->values[MEMORY_METER_SHARED] = NAN;
    this->values[MEMORY_METER_COMPRESSED] = NAN;
    this->values[MEMORY_METER_AVAILABLE] = NAN;
-   Platform_setMemoryValues(this);
-   if ((this->mode == GRAPH_METERMODE || this->mode == BAR_METERMODE) && !settings->showCachedMemory) {
-      this->values[MEMORY_METER_BUFFERS] = 0;
-      this->values[MEMORY_METER_CACHE] = 0;
-   }
-   /* Do not print available memory in bar mode */
-   static_assert(MEMORY_METER_AVAILABLE + 1 == MEMORY_METER_ITEMCOUNT,
-      "MEMORY_METER_AVAILABLE is not the last item in MemoryMeterValues");
-   this->curItems = MEMORY_METER_AVAILABLE;
-
-   /* we actually want to show "used + shared + compressed" */
-   double used = this->values[MEMORY_METER_USED];
-   if (isPositive(this->values[MEMORY_METER_SHARED]))
-      used += this->values[MEMORY_METER_SHARED];
-   if (isPositive(this->values[MEMORY_METER_COMPRESSED]))
-      used += this->values[MEMORY_METER_COMPRESSED];
-
-   written = Meter_humanUnit(buffer, used, size);
+
+   double usedNumber = 0;
+   Platform_setMemoryValues(this, &usedNumber);
+
+   written = Meter_humanUnit(buffer, usedNumber, size);
```

Comment:
I can't exactly say I love this change of the function signature here, but given the background behind this change, we'd probably have to expose far to much logic here otherwise. I'll leave this for the others as a note.